balancer-js/src/modules/zaps/bbausd2-migrations/bbausd1.integration.spec.ts

```diff
@@ -0,0 +1,314 @@
+// WARNING: don't use hardhat SignerWithAddress to sendTransactions!!
+// It's not working and we didn't have time to figure out why.
+// Use JsonRpcSigner instead
+const impersonateAccount = async (account: string) => {
+  await provider.send('hardhat_impersonateAccount', [account]);
+  await setBalance(account, parseEther('10000'));
+  return provider.getSigner(account);
+};
+
+const signRelayerApproval = async (
+  relayerAddress: string,
+  signerAddress: string,
+  signer: JsonRpcSigner
+): Promise<string> => {
+  const approval = contracts.vault.interface.encodeFunctionData(
+    'setRelayerApproval',
+    [signerAddress, relayerAddress, true]
+  );
+
+  const signature =
+    await RelayerAuthorization.signSetRelayerApprovalAuthorization(
+      contracts.vault,
+      signer,
+      relayerAddress,
+      approval
+    );
+
+  const calldata = RelayerAuthorization.encodeCalldataAuthorization(
+    '0x',
+    MaxUint256,
+    signature
+  );
+
+  return calldata;
+};
+
+const reset = async () =>
+  provider.send('hardhat_reset', [
+    {
+      forking: {
+        jsonRpcUrl,
+        blockNumber: (blockNumber && parseInt(blockNumber)) || 7277540,
+      },
+    },
+  ]);
+
+const move = async (
+  token: string,
+  from: string,
+  to: string
+): Promise<BigNumber> => {
+  const holder = await impersonateAccount(from);
+  const balance = await getErc20Balance(token, from);
+  await contracts.ERC20(token, provider).connect(holder).transfer(to, balance);
+
+  return balance;
+};
+
+const stake = async (
+  signer: JsonRpcSigner,
+  balance: BigNumber
+): Promise<void> => {
+  await (
+    await contracts
+      .ERC20(poolAddress, provider)
+      .connect(signer)
+      .approve(gaugeAddress, MaxUint256)
+  ).wait();
+
+  await (
+    await signer.sendTransaction({
+      to: gaugeAddress,
+      data: liquidityGauge.encodeFunctionData('deposit', [balance]),
+    })
+  ).wait();
+};
+
+describe('bbausd migration execution', async () => {
+  let signer: JsonRpcSigner;
+  let signerAddress: string;
+  let authorisation: string;
+  let balance: BigNumber;
+  let pool: PoolModel;
+
+  beforeEach(async function () {
+    this.timeout(20000);
+
+    await reset();
+
+    signer = provider.getSigner();
+    signerAddress = await signer.getAddress();
+    authorisation = await signRelayerApproval(relayer, signerAddress, signer);
+    // Transfer tokens from existing user account to signer
+    // We need that to test signatures, because hardhat doesn't have impersonated accounts private keys
+    balance = await move(poolAddress, holderAddress, signerAddress);
+
+    const config = {
+      network,
+      rpcUrl,
+    };
+    const subgraph = new Subgraph(config);
+    const pools = new PoolsProvider(
+      config,
+      new SubgraphPoolRepository(subgraph.client)
+    );
+    const _pool = await pools.findBy('address', poolAddress);
+    if (!_pool) throw new BalancerError(BalancerErrorCode.POOL_DOESNT_EXIST);
+    pool = _pool;
```

Comment:
how about handling that without temp variables?
```suggestion
    await pools.findBy('address', poolAddress).then((res) => {
      if (!res) throw new BalancerError(BalancerErrorCode.POOL_DOESNT_EXIST);
      pool = res;
    });
```

Reply:
with direct assignment TS is complaining about an undefined value:
`pool = await pools.findBy('address', poolAddress);`